torch_xla/csrc/ops/dynamic_ir.cpp

```diff
@@ -23,95 +23,78 @@ XlaOpVector SizeNode::Lower(LoweringContext* loctx) const {
 }
 
 int64_t SizeNode::getStaticValue() const {
-  return operand(0).node->shape(0).size(dim_);
-}
-
-bool SizeNode::isSymbolic() const {
-  auto symbolic_vec = operand(0).node->shape(0).is_symbolic();
-  if (!symbolic_vec.has_value()) {
-    return true;
-  }
-  return symbolic_vec->at(dim_);
+  // Not all IR has torch::lazy::shape now, use xla::shape to unblock
+  // the development.
+  return dynamic_cast<const XlaNode*>(operand(0).node)
+      ->xla_shape(operand(0).index)
+      .dimensions(dim_);
 }
 
 std::string SizeNode::ToString() const { return "SizeNode"; }
 
 SizeAdd::SizeAdd(torch::lazy::Value a, torch::lazy::Value b)
     : XlaNode(torch::lazy::OpKind{c10::Symbol::fromQualString("aten::add")},
-              {a, b}, xla::ShapeUtil::MakeShape(xla::S32, {}), 1){};
+              {a, b}, xla::ShapeUtil::MakeShape(xla::S32, {}), 1) {
+  // SizeAdd can only be perfomed between two DimensionNode
+  XLA_CHECK(DimCast(operand(0)));
+  XLA_CHECK(DimCast(operand(1)));
+};
 
 int64_t SizeAdd::getStaticValue() const {
-  return dynamic_cast<const torch::lazy::DimensionNode*>(operand(0).node)
-             ->getStaticValue() +
-         dynamic_cast<const torch::lazy::DimensionNode*>(operand(1).node)
-             ->getStaticValue();
-}
-
-bool SizeAdd::isSymbolic() const {
-  return DimCast(operand(0))->isSymbolic() || DimCast(operand(1))->isSymbolic();
+  return DimCast(operand(0))->getStaticValue() +
+         DimCast(operand(1))->getStaticValue();
 }
 
 std::string SizeAdd::ToString() const { return "SizeAdd"; }
 
 XlaOpVector SizeAdd::Lower(LoweringContext* loctx) const {
   auto input1 = loctx->GetOutputOp(operand(0));
   auto input2 = loctx->GetOutputOp(operand(1));
-  return ReturnOp(
-      (xla::GetDimensionSize(input1, 0) + xla::GetDimensionSize(input2, 0)),
-      loctx);
+  return ReturnOp((input1 + input2), loctx);
```

Comment:
okay this makes more sense to me since I wasn't sure why we needed to do `GetDimensionSize` on scalars 😸